internal/yggdrasil/yggdrasil.go

```diff
@@ -272,46 +275,23 @@ func (h *Handler) PostDataMessageForDevice(ctx context.Context, params yggdrasil
 	msg := params.Message
 	switch msg.Directive {
 	case "heartbeat":
-		heartbeat := models.Heartbeat{}
+		hb := models.Heartbeat{}
 		contentJson, _ := json.Marshal(msg.Content)
-		err := json.Unmarshal(contentJson, &heartbeat)
+		err := json.Unmarshal(contentJson, &hb)
 		if err != nil {
 			return operations.NewPostDataMessageForDeviceBadRequest()
 		}
-		logger.V(1).Info("received heartbeat", "content", heartbeat)
-		edgeDevice, err := h.deviceRepository.Read(ctx, params.DeviceID, h.initialNamespace)
+		err = h.heartbeatHandler.Process(ctx, heartbeat.Notification{
+			DeviceID:  deviceID,
+			Namespace: h.initialNamespace,
+			Heartbeat: &hb,
+		})
 		if err != nil {
 			if errors.IsNotFound(err) {
+				logger.V(1).Info("Device not found")
 				return operations.NewPostDataMessageForDeviceNotFound()
 			}
-			return operations.NewPostDataMessageForDeviceInternalServerError()
-		}
-
-		// Produce k8s events based on the device-worker events:
-		events := heartbeat.Events
-		for _, event := range events {
-			if event == nil {
-				continue
-			}
-
-			if event.Type == models.EventInfoTypeWarn {
-				h.recorder.Event(edgeDevice, corev1.EventTypeWarning, event.Reason, event.Message)
-			} else {
-				h.recorder.Event(edgeDevice, corev1.EventTypeNormal, event.Reason, event.Message)
-			}
-		}
-
-		err = h.updateDeviceStatus(ctx, edgeDevice, func(device *v1alpha1.EdgeDevice) {
```

Comment:
You deleted this, but you shouldn't no?

Reply:
Nope, it's good. I haven't adjusted a few tests, though.